src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -2604,15 +2606,15 @@ private WorkQueue submissionQueue(int r) {
                 if (w == null)
                     w = new WorkQueue(null, id, 0, false);
                 w.phase = id;
-                long stop = lockRunState() & STOP;
-                if (stop == 0L && queues == qs && qs[i] == null) {
+                long shutdown = lockRunState() & SHUTDOWN;
+                if (shutdown == 0L && queues == qs && qs[i] == null) {
                     q = qs[i] = w;                   // else retry
                     w = null;
                 }
                 unlockRunState();
                 if (q != null)
                     return q;
-                if (stop != 0L)
+                if (shutdown != 0L)
```

Comment:
@DougLea Any benefit of doing this vs `boolean shutdown = (lockRunState() & SHUTDOWN) != 0L` and check `!shutdown` and `shutdown`

Reply:
It doesn't seem to matter with respect to generated code. I did it this way for symmetry to other similar usages (including tryTerminate) but for more symmetry should have named it isShutdown, so just changed it.